crates/sui-proxy/src/middleware.rs

```diff
@@ -27,6 +27,25 @@ static MIDDLEWARE_OPS: Lazy<CounterVec> = Lazy::new(|| {
     .unwrap()
 });
 
+static MIDDLEWARE_HEADERS: Lazy<CounterVec> = Lazy::new(|| {
+    register_counter_vec!(
+        "middleware_headers",
+        "Operations counters and status for axum middleware.",
+        &["header", "value"]
+    )
+    .unwrap()
+});
+
+/// we expect sui-node to send us an http header content-length encoding.
+pub async fn expect_content_length<B>(
+    TypedHeader(content_length): TypedHeader<ContentLength>,
+    request: Request<B>,
+    next: Next<B>,
+) -> Result<Response, (StatusCode, &'static str)> {
+    MIDDLEWARE_HEADERS.with_label_values(&["content-length", &format!("{}", content_length.0)]);
+    Ok(next.run(request).await)
```

Comment:
Does this need to be a result if we never return the `Err` variant?

Reply:
yes, checkout the middleware docs for axum and hyper to see why. it's similar to a nodejs express.js nexting pattern.  we must return a response or an error and it must be in the above style.

Reply:
Sounds good. It's strange that the endpoint API is relatively flexible but the middleware one is stricter but 🤷‍♂️. Thanks for the response.